### PR TITLE
Split Nvidia recipe and creation of nvidia_driver resource

### PR DESCRIPTION
**Pull request by francesco-giordano**
### Description of changes
* Split of the recipe nvidia in 4 recipes: nvidia_driver, cuda, fabric_manager and gdrcopy.
* Transform nvidia_driver in a resource
* Moved the spec test from config to install

### Tests
* Tested locally with x86 and arm64 instances

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cookbooks/aws-parallelcluster-install/resources/nvidia_driver/partial/_nvidia_driver_common.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+unified_mode true
+default_action :setup
+
+return unless node['cluster']['nvidia']['enabled'] == 'yes' || node['cluster']['nvidia']['enabled'] == true
```
Comment (enrico-usai) [inline cookbooks/aws-parallelcluster-install/resources/nvidia_driver/partial/_nvidia_driver_common.rb]: I'd move this check into the `setup` action, in this way the action is defined and the resource definition is not broken.

File: kitchen.resources-install.yml [hunk 2]
```diff
@@ -171,3 +171,21 @@ suites:
       resource: system_authentication
       dependencies:
         - resource:package_repos
+  # nvidia_driver can be executed only on a graphic EC2 instance example: g5.xlarge or g5g.xlarge
```
Comment (enrico-usai) [inline kitchen.resources-install.yml]: you can pass:
``` 
    driver:
      instance_type: g5.xlarge 
```

to the test and move the comment there.

**Review by enrico-usai — COMMENTED** [review]
Great job!

File: kitchen.resources-install.yml (hunk 2 above)
Comment (francesco-giordano) [reply kitchen.resources-install.yml]: Done

File: cookbooks/aws-parallelcluster-install/resources/nvidia_driver/partial/_nvidia_driver_common.rb (hunk 1 above)
Comment (francesco-giordano) [reply cookbooks/aws-parallelcluster-install/resources/nvidia_driver/partial/_nvidia_driver_common.rb]: Done

**Review by enrico-usai — APPROVED** [review]

**Pull request merged**